src/Common/TLDListsHolder.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <common/defines.h>
+#include <common/StringRef.h>
+#include <Common/HashTable/StringHashSet.h>
+#include <Common/Arena.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace DB
+{
+
+/// Custom TLD List
+///
+/// Unlike tldLookup (which uses gperf) this one uses plain StringHashSet.
+class TLDList
+{
+public:
+    using Container = StringHashSet<>;
+
+    TLDList(size_t size);
+
+    /// Return true if the tld_container does not contains such element.
+    bool insert(const StringRef & host);
+    /// Check is there such TLD
+    bool has(const StringRef & host) const;
+    size_t size() const { return tld_container.size(); }
+
+private:
+    Container tld_container;
+    std::unique_ptr<Arena> pool;
+};
+
+class TLDListsHolder
+{
+public:
+    /// FIXME: use dense_hash_map and better hash function
```

Comment:
This is for storing the list of custom TLD lists, it's small and we search it once per block, so probably any map will do?

Reply:
Yep, sure, that's why it is just a comment (I leave this comment only because unordered_set is way more less optimal then dense_hash_map and I couldn't resist)